src/lib/mina_base/transaction_logic.ml

```diff
@@ -2607,7 +2607,7 @@ module For_tests = struct
                     ; token_id = Token_id.default
                     ; balance_change =
                         Amount.Signed.(negate (of_unsigned amount))
-                    ; increment_nonce = true (* TODO(#9743) *)
+                    ; increment_nonce = true
```

Comment:
Nit: don't we want this to be `increment_nonce= false; use_full_commitment= true`, to avoid double-incrementing the nonce?

Reply:
I'll write another test for that? I think we should have both  (use full commitment or increment nonce twice) since they are both are supported